include/swift/AST/DiagnosticsSema.def

```diff
@@ -2485,7 +2485,10 @@ ERROR(tuple_pattern_in_non_tuple_context,none,
       "tuple pattern cannot match values of the non-tuple type %0", (Type))
 ERROR(closure_argument_list_tuple,none,
       "contextual closure type %0 expects %1 argument%s1, "
-      "but %2 were used in closure body", (Type, unsigned, unsigned))
+      "but %2 %select{was|were}3 used in closure body", (Type, unsigned, unsigned, bool))
+ERROR(closure_argument_list_single_tuple,none,
+      "contextual closure type specifies %0, but %1 %select{was|were}2 used in closure body, "
+      "try add extra parenthesis around the single tuple argument", (Type, unsigned, bool))
```

Comment:
my bad. I mis-read your comment as the change on line:2488 since they are quoted as one green block. I would add it, thanks!